worldengine/draw.py

```diff
@@ -161,14 +163,47 @@ def __setitem__(self, item, value):
         self.pixels[item] = value
 
 
-def draw_simple_elevation(data, width, height, sea_level, target):
+def draw_simple_elevation(world, sea_level, target):
     """ This function can be used on a generic canvas (either an image to save
         on disk or a canvas part of a GUI)
     """
-    for y in range(height):
-        for x in range(width):
-            e = data[y][x]#TODO: numpy-notation [y,x] is a little faster; all passed arrays would have to be numpy-arrays
-            r, g, b = elevation_color(e, sea_level)
+    min_elev_sea = None
+    max_elev_sea = None
+    min_elev_land = None
+    max_elev_land = None
+    for y in range(world.height):
+        for x in range(world.width):
+            e = world.elevation['data'][y,x]
+            if sea_level is None:
+                if min_elev_land is None or e < min_elev_land:
+                    min_elev_land = e
+                if max_elev_land is None or e > max_elev_land:
+                    max_elev_land = e
+            elif world.is_land((x, y)):
+                if min_elev_land is None or e < min_elev_land:
+                    min_elev_land = e
+                if max_elev_land is None or e > max_elev_land:
+                    max_elev_land = e
+            else:
+                if min_elev_sea is None or e < min_elev_sea:
+                    min_elev_sea = e
+                if max_elev_sea is None or e > max_elev_sea:
+                    max_elev_sea = e
+
+    elev_delta_land = (max_elev_land - min_elev_land)/11
+    if sea_level != None:
+        elev_delta_sea = max_elev_sea - min_elev_sea
+    
+    for y in range(world.height):
```

Comment:
If you added some numpy-code before this loop, you could get rid of the if-else-statements (and prepare the code for later changes where maybe the whole loop can be worked off in a single line of code):
```python
c = numpy.array(world.elevation['data'].shape)
c[numpy.logical_not(world.ocean)] = ((world.elevation['data'] - min_elev_land) / elev_delta_land) + 1)
c[world.ocean] = ((world.elevation['data'] - min_elev_sea) / elev_delta_sea))
```

Then loop over c and use ```c[x, y]```.